README.md

```diff
@@ -1,29 +1,288 @@
+  | Description       | `QW tool for ${repository_name}`                            |
+  | Repository access | `Only select repositories` -> Select the project repository |
+  | Permissions       | Only add repository permissions, see next table             |
+
+  | repository permissions | value          |
+  | ---------------------- | -------------- |
+  | Actions                | Read and write |
+  | Contents               | Read-only      |
+  | Issues                 | Read and write |
+  | Metadata               | Read-only      |
+  | Pull Requests          | Read and write |
+
+  Then click the `Generate token` button and copy the token (you can't get its value again if you close the page).
+
+- Add the personal access token to your machine's keychain (uses [keyring](https://pypi.org/project/keyring/))
+
+  ```shell
+  qw login
+  ```
+
+  <!--
+     - [name=Stef] For this I guess we should implement it in keyring using: service=`qw:github.com` username=`owner/repo`?
+     - [name=Stef] Should try and get a windows user early on to test that keyring works as expected
+  -->
+
+  > Please copy the personal access token and hit enter:
+
+  After you've done this, the system will check the personal access token. If you've configured for the remote URL <https://github.com/stefpiatek/dummy-medical-software> you would see:
+
+  > INFO: Can access "stefpiatek/dummy-medical-software"
+
+### Setup for a new user, on an existing project that uses QW
+
+The local and github repository have already been configured to work with `qw`, you'll just need to [add the personal access token](#setup-personal-access-token)
+
+### Customising configuration
+
+QW initialises with a `System` component for design inputs (used for generating a unique identifier, short code value is `X`).
+You can add extra components to the configuration using qw:
+
+```shell
+qw config add-service --name="Drug dosage" --short-code=D
+```
+
+> INFO: Added "Drug dosage (D)" to local compenents, please commit this file
+> INFO: Added tag "qw-component-d" to github
+
+## Configuration using Gitlab
+
+Intentionally left blank at the moment for brevity. Will aim for being able to implement this.
+
+# Using QW with github
+
+QW uses existing issues and pull requests to track the different design and development stages along with managing risks and their mitigations.
+
+## Creating QW items
+
+### User needs
+
+- In your github repository, Add a new issue, selecting the User needs template
+  ![](https://hackmd.io/_uploads/Syo1w1oy6.png)
+- Fill out the required fields and any other information if it exists
+  ![](https://hackmd.io/_uploads/rJG7_1okT.png)
+- Hit `Submit new issue` and the issue will be rendered like this: ![](https://hackmd.io/_uploads/BktGcJo1p.png)
+- The first comment of this issue will be used by `qw` for tracking, but you can edit anything manually after the `Other information` header and this will not interfere with the tool
+
+### Design inputs
+
```

Comment:
yeah I grabbed them from the types defined in the ISO 62304 standard, but did wonder if there'd be others

Reply:
already seen one, `performance`. Hmmm going to have to make this configurable aren't we

Reply:
62304 has a list (which it describes as non-overlapping):
Functional and Capability (includes performance), Inputs and Outputs (I think this means file formats and the like), Interfaces to other systems, Alarms/warnings/messages (I think this means that we are required to produce, not respond to), Security, User Interfaces, Data definition and database (not sure, the standard just gives "form", "fit" and "function" as examples, I don't see why this isn't Inputs and Outputs), Installation and Maintenance (done by us), IT Network, User Maintenance (done by them), Regulatory.
Risk Control Measures are given a separate subclause, but it seems to me these function as normal requirements, but they are important to the Risk Control aspect we aren't too worried about yet.
I don't know whether we should follow this, but if we do it will be easier for the devs to demonstrate that they have met clause 5.2.2 that says they should have considered each of these.

Reply:
Ah I see you already have this!

Reply:
Performance is included in "Functional and Capability". The advantage of following their schema is that we can extract them into a document (at some point!) that demonstrates how 5.2.2 has been met. We can warn for items that have no requirements. The fix for this is either add the tag to appropriate requirements (we have to allow people to add multiple if they like because 5.2.2 states that these are non-overlapping), or provide some text (nice MD document again?) providing text as to why this type of requirement does not feature.

Reply:
Agreed.